posthog/api/decide.py

```
@@ -336,3 +292,53 @@ def get_decide(request: HttpRequest):
+            sample_rate = team.session_recording_sample_rate or None
+            if sample_rate == "1.00":
+                sample_rate = None
+
+            minimum_duration = team.session_recording_minimum_duration_milliseconds or None
+
+            linked_flag = None
+            linked_flag_config = team.session_recording_linked_flag or None
+            if isinstance(linked_flag_config, Dict):
+                linked_flag_key = linked_flag_config.get("key", None)
+                linked_flag_variant = linked_flag_config.get("variant", None)
+                if linked_flag_variant is not None:
+                    linked_flag = {"flag": linked_flag_key, "variant": linked_flag_variant}
+                else:
+                    linked_flag = linked_flag_key
+
+            session_recording_config_response = {
+                "endpoint": "/s/",
+                "consoleLogRecordingEnabled": capture_console_logs,
+                "recorderVersion": "v2",
+                "sampleRate": sample_rate,
+                "minimumDurationMilliseconds": minimum_duration,
+                "linkedFlag": linked_flag,
+                "networkPayloadCapture": team.session_recording_network_payload_capture_config or None,
+            }
+
+            if isinstance(team.session_replay_config, Dict):
+                record_canvas = team.session_replay_config.get("record_canvas", False)
+                session_recording_config_response.update(
+                    {
+                        "recordCanvas": record_canvas,
+                        # hard coded during beta while we decide on sensible values
+                        "canvasFps": 4 if record_canvas else None,
+                        "canvasQuality": "0.6" if record_canvas else None,
+                    }
+                )
+    except Exception as e:
+        capture_exception(e)  # we don't want to fail decide if session recording config fails to load
```

swallow but report any error